src/sage/modular/modform/element.py

```diff
@@ -3544,18 +3544,25 @@ def _mul_(self, other):
         TESTS::
 
             sage: M = ModularFormsRing(1)
-            sage: f4 = ModularForms(1, 4).0; f6 = ModularForms(1, 6).0;
-            sage: F4 = M(f4); F6 = M(f6);
+            sage: F4 = M.0; F6 = M.1;
             sage: F4*F6 # indirect doctest
             1 - 264*q - 135432*q^2 - 5196576*q^3 - 69341448*q^4 - 515625264*q^5 + O(q^6)
+
+        This shows that the issue at :issue:`35932` is fixed::
+
+            sage: (F4 + M(1))^2
+            4 + 960*q + 66240*q^2 + 1063680*q^3 + 7961280*q^4 + 37560960*q^5 + O(q^6)
         """
         GM = self.__class__
         f_self = self._forms_dictionary
         f_other = other._forms_dictionary
         f_mul = {}
```

Comment:
try https://docs.python.org/3/library/collections.html#collections.defaultdict

Reply:
Thanks for the feedback! I have done that, but I am not sure if I have to cast back to `dict` at the end or if it is fine to leave it as `defaultdict`.

Reply:
It's fine to leave it as is. The module element constructor accepts it just fine